### PR TITLE
[Feature] Add shortcuts manager to print layouts

### LINKED ISSUE #25011: Customize composer keyboard shortcuts
Author Name: **Juraj Betak** (@jurajbetak)
Original Redmine Issue: [17112](https://issues.qgis.org/issues/17112)

Redmine category:map_composer/printing
Assignee: Nyall Dawson

---
Settings - Configure shortcuts  … does not include composer tools.
Is it possible to customize it here or some other place?

### LINKED ISSUE #41610: Unable to change shortcut keys for actions in the layout view (print composer).
**Describe the bug**
Actions available in "Print Composer" are missing in the "Keyboard Shortcuts" list. Even the actions to which hotkeys are assigned are missing. Some actions whose names coincide with the names of actions in the main menu can be found in the "Keyboard Shortcuts" list (for example "Refresh - F5").

**How to Reproduce**

1. Go to 'Settings' =>  'Keyboard Shortcuts'.
2. Search any actions available in the 'Print Composer' menu (no results found).

5uOG9nrEj0

**QGIS and OS versions**

QGIS version
3.14.16-Pi
QGIS code revision
df27394552
Compiled against Qt
5.11.2
Running against Qt
5.11.2
Compiled against GDAL/OGR
3.0.4
Running against GDAL/OGR
3.0.4
Compiled against GEOS
3.8.1-CAPI-1.13.3
Running against GEOS
3.8.1-CAPI-1.13.3
Compiled against SQLite
3.29.0
Running against SQLite
3.29.0
PostgreSQL Client Version
11.5
SpatiaLite Version
4.3.0
QWT Version
6.1.3
QScintilla2 Version
2.10.8
Compiled against PROJ
6.3.2
Running against PROJ
Rel. 6.3.2, May 1st, 2020
OS Version
Windows 10 (10.0)
Active python plugins
AppendFeaturesToLayer; 
DigitizingTools; 
ee_plugin; 
FreehandRasterGeoreferencer; 
Go2NextFeature3; 
latlontools; 
mapswipetool_plugin; 
minimum_spanning_tree; 
qgis_gee_data_catalog; 
qgsAzimuth; 
sentinel2_download; 
db_manager; 
MetaSearch; 
processing

### PR DESCRIPTION
**Pull request by YoannQDQ**
- Fixes #41610
- Fixes #25011

## Description

QGIS main window has a shortcut manager, accessible from the settings menu, that allow to configure and override existing keyboard shortcuts (`QShortcut` and `QAction`).

This PR adds a similar keyboard manager ([`QgsShortcutsManager`](https://api.qgis.org/api/classQgsShortcutsManager.html)) and its configuration dialog ([`QgsConfigureShortcutsDialog`](https://api.qgis.org/api/classQgsConfigureShortcutsDialog.html)) to the Print Layout interface.

shortcuts_layout

### CODE REVIEW

File: python/gui/auto_generated/layout/qgslayoutdesignerinterface.sip.in [hunk 1]
```diff
@@ -378,6 +378,13 @@ designer is in report mode.
 
 
 .. versionadded:: 3.4
+%End
+
+    virtual QgsShortcutsManager *shortcutsManager() = 0;
```
Comment (nyalldawson) [inline python/gui/auto_generated/layout/qgslayoutdesignerinterface.sip.in:383]: Does this need to be exposed? I can't see how a plugin would make use of it. (You can add SIP_SKIP to the declaration line to omit it)

Comment (YoannQDQ) [reply python/gui/auto_generated/layout/qgslayoutdesignerinterface.sip.in:383]: Yeah, I wasn't too sure about adding it to the QgsLayoutDesignerInterface... I think I will just discard the second commit of this PR.

Comment (nyalldawson) [reply python/gui/auto_generated/layout/qgslayoutdesignerinterface.sip.in:383]: I'd remove it -- if we expose them it becomes part of stable API and we lose flexibility in future.

Comment (YoannQDQ) [reply python/gui/auto_generated/layout/qgslayoutdesignerinterface.sip.in]: Done!

**Review by nyalldawson — APPROVED** [review]

File: src/app/layout/qgslayoutdesignerdialog.h [hunk 2]
```diff
@@ -49,6 +49,7 @@ class QgsFeature;
 class QgsMasterLayoutInterface;
 class QgsLayoutGuideWidget;
 class QgsScreenHelper;
+class QgsConfigureShortcutsDialog;
```
Comment (nyalldawson) [inline src/app/layout/qgslayoutdesignerdialog.h:52]:
```suggestion
class QgsConfigureShortcutsDialog;
class QgsShortcutsManager;
```

**Pull request merged**